haskell/src/Monocle/Api/Server.hs

```diff
@@ -299,6 +311,120 @@ crawlerCommitInfo request = do
         . Enumerated
         $ Right err
 
+-- | /task_data endpoints
+taskDataTaskDataAdd :: TaskDataPB.TaskDataAddRequest -> AppM TaskDataPB.TaskDataAddResponse
+taskDataTaskDataAdd TaskDataPB.TaskDataAddRequest {..} = do
+  tenants <- getConfig
+  let requestE = do
+        index <-
+          Config.lookupTenant tenants (toStrict taskDataAddRequestIndex)
+            `orDie` TaskDataPB.TaskDataCommitErrorUnknownApiKey
+
+        void $
+          Config.lookupCrawler index (toStrict taskDataAddRequestCrawler)
+            `orDie` TaskDataPB.TaskDataCommitErrorUnknownCrawler
+
+        when
+          (Config.crawlers_api_key index /= Just (toStrict taskDataAddRequestApikey))
+          (Left TaskDataPB.TaskDataCommitErrorUnknownApiKey)
+
+        pure index
```

Comment:
nit: it might be worth adding a common `validateTaskDataRequest` similar to the existing `validateSearchRequest`

Reply:
Sure will do.